src/androidMain/kotlin/com/liftric/job/queue/NetworkManager.kt

```diff
@@ -0,0 +1,55 @@
+package com.liftric.job.queue
+
+import android.annotation.SuppressLint
+import android.annotation.TargetApi
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+@SuppressLint("MissingPermission")
+actual class NetworkManager(context: Context) {
+    private var connectivityManager: ConnectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    private val connectionPublisher = MutableStateFlow(getCurrentNetworkConnection())
+
+    fun observeNetworkConnection(): Flow<NetworkState?> = connectionPublisher
```

Comment:
Do I understand the correctly that on `iOS` we basically observe the connection, hence the `Flow` could emit changes and on `Android` we only get the state at the time of the function invocation?

Reply:
Well, I just realized that now and I guess the answer is yes. I will do the following change:

```suggestion
     val network: Flow<NetworkState?> = connectionPublisher
```

and then listen it just as in `IOS` in Android `NetworkListener`

```
override fun observeNetworkState() {
        scope.launch {
            networkManager
                .network
                .collectLatest { currentNetworkState ->
                    _currentNetworkState.emit(currentNetworkState ?: NetworkState.NONE)
                }
        }
    }
```